### PR TITLE
[autom8] Add comprehensive process management capabilities to the...

### PR DESCRIPTION
**Pull request by HammadTheOne**
## Summary

Add comprehensive process management capabilities to the autom8 GUI, allowing users to monitor, control, and inspect Claude CLI processes spawned during implementation runs.

Currently, the GUI's Active Runs tab displays session cards with basic information (state, progress, last 12 lines of output) but offers no way to interact with running processes. Users cannot kill a runaway process, pause work to review changes, or inspect detailed process information without resorting to the terminal.

This feature adds: (1) process lifecycle controls (kill, pause, resume) accessible from expandable session cards, (2) real-time resource monitoring (CPU, memory, PID), (3) a detail modal with full scrollable output history and error information, and (4) the underlying infrastructure to track process metadata that doesn't currently exist.

### US-001: Add PID and Process Metadata Tracking

Extend `ClaudeRunner` to capture and expose process metadata when spawning the Claude CLI subprocess.

**Acceptance Criteria:**

- [x] `ClaudeRunner` stores the child process PID via `child.id()` after successful spawn
- [x] PID is accessible via a new `pub fn pid(&self) -> Option<u32>` method
- [x] PID is cleared when process terminates (kill, normal exit, or error)
- [x] Add `ProcessInfo` struct to hold PID and spawn timestamp
- [x] Existing tests pass; add unit tests for PID tracking

**Notes:**

The PID is needed for resource monitoring and display in the UI. Store spawn time for duration calculations.

### US-002: Implement Pause and Resume Signals

Add SIGSTOP (pause) and SIGCONT (resume) signal support to `ClaudeRunner` for process control.

**Acceptance Criteria:**

- [x] Add `pub fn pause(&self) -> Result<bool>` that sends SIGSTOP to the child process
- [x] Add `pub fn resume(&self) -> Result<bool>` that sends SIGCONT to the child process
- [x] Both methods return `Ok(true)` if signal sent, `Ok(false)` if no process running
- [x] Add `pub fn is_paused(&self) -> bool` method to track pause state
- [x] Handle edge cases: pausing already-paused process, resuming non-paused process
- [x] Use `libc::kill()` with appropriate signals (Unix-specific, compile-gate for non-Unix)
- [x] Add unit tests for pause/resume functionality

**Notes:**

On non-Unix platforms (Windows), these methods should return an error or no-op gracefully. The pause state needs to be tracked internally since there's no way to query it from the OS.

### US-003: Add Resource Monitoring Infrastructure

Integrate the `sysinfo` crate to monitor CPU and memory usage of spawned Claude processes.

**Acceptance Criteria:**

- [x] Add `sysinfo` crate to `Cargo.toml`
- [x] Create `ProcessMetrics` struct with fields: `cpu_percent: f32`, `memory_bytes: u64`, `memory_percent: f32`
- [x] Create `ProcessMonitor` struct that wraps `sysinfo::System` and tracks a specific PID
- [x] Add `pub fn refresh(&mut self)` to update metrics for the tracked process
- [x] Add `pub fn metrics(&self) -> Option<ProcessMetrics>` to get current metrics
- [x] Handle process-not-found gracefully (process may have exited between checks)
- [x] Add to a new file: `src/process/monitor.rs` (create `process` module)

**Notes:**

`sysinfo` is cross-platform and handles the OS-specific details. Refresh should be cheap enough to call at 500ms intervals.

### US-004: Increase Live Output History Cap

Increase the output history buffer from 50 to 500 lines and ensure efficient storage/retrieval.

**Acceptance Criteria:**

- [x] Change `LIVE_STATE_MAX_LINES` constant from 50 to 500
- [x] Update `LiveState` serialization to handle larger payloads efficiently
- [x] Verify `live.json` file size remains reasonable (~50KB max for 500 lines)
- [x] Add `output_line_count: usize` field to `LiveState` for quick count without loading all lines
- [x] Existing monitor command and GUI continue to work with larger buffer
- [x] No performance regression in write frequency (still ~200ms intervals)

**Notes:**

500 lines at ~100 chars average = ~50KB, which is acceptable. The GUI modal will display all lines with scrolling.

### US-005: Add Expandable Session Cards with Action Buttons

Extend the Active Runs session cards to be expandable, revealing process controls and additional metadata.

**Acceptance Criteria:**

- [x] Add expand/collapse toggle button (chevron icon or similar) to each session card header
- [x] Collapsed state (default): Current display (state dot, progress, duration, 12 lines output)
- [x] Expanded state adds: Process info section: PID, spawn time, pause state indicator
- [x] Expanded state adds: Resource meters: CPU % bar, Memory usage (MB and %)
- [x] Expanded state adds: Action buttons row: Kill (red), Pause/Resume (toggle), View Details (opens modal)
- [x] Smooth expand/collapse animation (egui's animation support)
- [x] Expanded state persists during refresh cycles (don't collapse on data reload)
- [x] Action buttons are disabled appropriately (e.g., can't pause an already-paused process)
- [x] Kill button shows confirmation before executing

**Notes:**

Keep the card width consistent; expansion adds height. Use the existing theme colors for buttons (red for destructive, blue for info).

### US-006: Implement Process Detail Modal

Create a modal dialog for viewing comprehensive process details and full output history.

**Acceptance Criteria:**

- [x] Modal opens when clicking "View Details" button on expanded session card
- [x] Modal header: Session name, current state badge, duration
- [x] Modal body sections: Process Info with PID, spawn timestamp, pause state, exit code (if terminated)
- [x] Modal body sections: Resources with CPU and memory with real-time updating gauges/graphs
- [x] Modal body sections: Output History with full scrollable output (all 500 lines), monospace font, auto-scroll toggle
- [x] Modal body sections: Error Details showing exit code, stderr output, error message if failed/errored
- [x] Modal footer: Action buttons (Kill, Pause/Resume, Close)
- [x] Modal is dismissible via Close button, Escape key, or clicking outside
- [x] Output auto-scrolls to bottom by default; user can scroll up to review history
- [x] Scrolling up disables auto-scroll; button to re-enable and jump to bottom
- [x] Modal size: ~80% of window width, ~90% of window height (responsive)

**Notes:**

Use egui's `Window` or `Modal` pattern. The output view is the main focus—make it take most of the vertical space.

### US-007: Wire Up Real-Time Resource Updates

Connect the resource monitoring infrastructure to the GUI refresh cycle for real-time updates.

**Acceptance Criteria:**

- [x] `ProcessMonitor` instances are created/tracked per active session
- [x] Resource metrics refresh on the same 500ms interval as other GUI data
- [x] Metrics are passed through the shared data layer (`SessionData` in `ui/shared/mod.rs`)
- [x] Add `process_metrics: Option<ProcessMetrics>` field to `SessionData`
- [x] GUI reads metrics from `SessionData` and renders in cards/modal
- [x] Monitors are cleaned up when sessions end (no leaked resources)
- [x] Handle race condition: process exits between metric refresh and render

**Notes:**

Consider whether to run monitoring in the main thread or spawn a background task. Given the 500ms interval and `sysinfo`'s efficiency, main thread should be fine.

### US-008: Implement Kill/Pause/Resume Actions in GUI

Wire up the action buttons to actually control the processes.

**Acceptance Criteria:**

- [ ] Kill button calls appropriate backend to terminate the session's Claude process
- [ ] After kill: session state updates to show interrupted/failed status
- [ ] Pause button sends SIGSTOP; button toggles to show "Resume"
- [ ] Resume button sends SIGCONT; button toggles back to "Pause"
- [ ] Paused state is visually indicated (e.g., "Paused" badge, dimmed output area)
- [ ] Actions provide feedback: brief "Killing..." / "Pausing..." state or toast notification
- [ ] Error handling: show user-friendly message if action fails
- [ ] Kill confirmation dialog: "Are you sure you want to kill this session? This cannot be undone."

**Notes:**

The runner's `ClaudeRunner` is not directly accessible from the GUI. Need to either: (a) expose an IPC/command mechanism, (b) send signals directly via PID, or (c) write a sentinel file the runner checks. Option (b) is simplest for kill/pause/resume since we have the PID.

### CODE REVIEW

File: src/claude/runner.rs [hunk 1]
```diff
@@ -26,21 +26,168 @@ const COMPLETION_SIGNAL: &str = "<promise>COMPLETE</promise>";
...
+            }
+        } else {
+            Ok(false)
+        }
+    }
+
+    /// Pauses the subprocess (non-Unix stub).
+    ///
+    /// On non-Unix platforms, this operation is not supported.
+    #[cfg(not(unix))]
```
[the remark this answers is not in the document]
Comment (louisboilard) [reply src/claude/runner.rs]: add little more details to this, we already have a full cross platform mechanism (ctrl-c and resume), we could just leverage this it would probably save a few thousand lines of code also, we don't need unsafe code and direct libc calls. of course I would not send a signal  (ctrl-c), we just handle things and call exit in the code after explicitly calling cleanup code to make sure that we add everything that we need on disk to resume properly.

For the sysinfo crate usage i think it's great but we only end up displaying two numbers and we have like ~250 lines of code to display 2 numbers? (ProcessMonitor).
